snapshot/restore/local_fs_output.go

```diff
@@ -315,7 +319,13 @@ func write(targetPath string, r fs.Reader) error {
 
 	name := f.Name()
 
-	if err := iocopy.JustCopy(f, r); err != nil {
+	if sparse {
+		err = iocopy.JustCopy(&sparseWriter{f: f}, r)
```

Comment:
perhaps move this to `iocopy.CopySparse(w, r, blockSize)` ?

w can be *os.File or a smaller interface.

Reply:
I imagine we'd want CopySparse to take `io.Writer` as well (so that the interfaces align). At that point, shouldn't we just push the write logic down to the writer?

How about we introduce `SparseFile` in a new package, which implements `File` and custom `Writer`?

Reply:
Alright, the context needed to perform all the ops the way we want to is all available in `FileSystemOutput.copyFileContent()`, but not any lower in the call chain. I've introduced a new package `sparsefile`, much like `atomicfile`, that has `Write()` and internal `copySparse()` functions. For now, we call only one of `atomicfile.Write(...)` and `sparsefile.Write(...)`. I'd like to come back and refactor this code to support both features together.